src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -72,37 +75,36 @@ pub fn compile_targets<'a, 'cfg: 'a>(targets: &[(&'a Target, &'a Profile)],
     try!(links::validate(deps));
 
     let dest = if build_config.release {"release"} else {"debug"};
-    let root = if resolve.root() == pkg.package_id() {
-        pkg
-    } else {
-        deps.iter().find(|p| p.package_id() == resolve.root()).unwrap()
-    };
+    let root = deps.iter().find(|p| p.package_id() == resolve.root()).unwrap();
```

Comment:
I think this logic will need to be updated, the notion of a "root" package doesn't really exist any more, so those which are depending on it probably need to be updated.